app.py

```diff
@@ -95,6 +95,38 @@ def logout():
     return redirect("/")
 
 
+@app.route("/user/edit", methods=["GET", "POST"])
+def edit_profile():
+    """Update profile for current user."""
+    if not g.user:
+        flash("Access unauthorized.", "danger")
+        return redirect("/")
+
+    form = UserAddForm(obj=g.user)
+
+    if form.validate_on_submit():
+        user = User.authenticate(g.user.username, form.password.data)
+
+        if user:
+            try:
+                user.username = (form.username.data,)
+                user.email = (form.email.data,)
+
+                db.session.commit()
+
+                flash("Profile Updated", "success")
+                return redirect("/")
+
+            except IntegrityError:
```

Comment:
@blakes24 why are you only handling the `IntegrityError`?......Can't we refactor to handle all possible errors that could arise from the try block?....What do you think ?

Reply:
The Integrity error occurs when there is a unique violation so I used the try block for that so I could return an error message that lets the user know that the username is ready taken. I'm not sure what other errors could arise that require specific user feedback. WTForms validates the user input and `@app.errorhandler(Exception)` catches any database exceptions that aren't specifically handled and displays the error page. What other errors should I be handling?

Reply:
Oh ok.....  I realize you are handling all the other db exceptions with `@app.errorhandler(Exception)`. This implementation on the other hand silently and invisibly hides other error conditions that can otherwise be quickly detected and dispatched. Once an error occurs, you are greatly hampered in your troubleshooting by a lack of critical information. What was the error/exception class? What call or data structure was involved? What line of code did the error originate from?

In that regard you can refactor and log any othe exception that may occur with the related stack trace:
`@app.errorhandler(Exception)
def server_error(e):
    """Display error page."""
    """Log error message with stack trace"""
    app.logger.exception("An error occured", e)
    db.session().rollback()
    return render_template("error.html"), 500`

Reply:
Thanks, that helps a lot. I've never used error logging before.